milli/src/search/new/mod.rs

```diff
@@ -442,6 +448,34 @@ pub fn execute_search(
 
     let fields_ids_map = ctx.index.fields_ids_map(ctx.txn)?;
 
+    let docids = match vector {
+        Some(vector) => {
+            // return the nearest documents that are also part of the candidates.
+            let mut searcher = Searcher::new();
+            let hnsw = ctx.index.vector_hnsw(ctx.txn)?.unwrap_or_default();
+            let ef = hnsw.len().min(100);
```

Comment:
I don't understand the `min(100)`

Reply:
We will search for a hundred documents in the HNSW and filter afterward. I get the length of the HNSW data structure to make sure the array is not bigger than that otherwise the HNSW crate badly panics.